realm/src/main/java/io/realm/RealmResults.java

```diff
@@ -105,6 +105,14 @@ public E get(int rowIndex) {
     }
 
     /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int indexOf(Object o) {
+        throw new NoSuchMethodError();
```

Comment:
Perhaps include a text message with a short description along the lines of "indexOf not supported on RealmResults" or something similar. The generic exception with no text description might be a bit unexpected.